src/Symfony/Component/HttpFoundation/UriSigner.php

```diff
@@ -124,7 +125,7 @@ public function checkRequest(Request $request): bool
 
     private function computeHash(string $uri): string
     {
-        return base64_encode(hash_hmac('sha256', $uri, $this->secret, true));
+        return sodium_bin2base64(hash_hmac('sha256', $uri, $this->secret, true), SODIUM_BASE64_VARIANT_URLSAFE_NO_PADDING);
```

Comment:
Introducing a dependency on ext-sodium just to provide URL-safe base64 encoding is clearly not necessary. It is very easy to do it in userland (using the same `strtr` you use for the comparison)